### DISCUSSION TITLE
Process memory usage different reporting after upgrading node from v12.14.1 to v16.14.1

### DISCUSSION THREAD

**Discussion by oancear05** [nodejs/help#4922] [Q&A]
### Version

16.14.1

### Platform

Linux testing-box 5.10.76-linuxkit #1 SMP Mon Nov 8 10:21:19 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

### What steps will reproduce the bug?

Upgrade node from v12.14.1 to v16.14.1 

### How often does it reproduce? Is there a required condition?

All the time, no require conditions

### What is the expected behavior?

The HEAP memory value should be lower than RSS memory and the RSS memory should increase consistently with the HEAP memory.
HEAP memory value should be the same behavior as it is before upgrading the node version to v16.14.1 

### What do you see instead?

After the node upgrade, the HEAP memory value is way higher than RSS one (please check the attached screenshot). I think it might be a memory leak or a bad reporting, but please let us know why I'm facing this memory increase. The data from the graphics was collected with the _**process.memoryUsage()**_ method. 
![image (30)](https://user-images.githubusercontent.com/89182319/164215043-a4f40ccd-0867-495a-b8e7-983b7cdcca2c.png)
 

![image (33)](https://user-images.githubusercontent.com/89182319/164215467-b2273014-d597-411e-9daa-73218275763e.png)

**Comment by bnoordhuis** [comment]
> The HEAP memory value should be lower than RSS memory and the RSS memory should increase consistently with the HEAP memory.

You may have observed that with older node versions but that relation doesn't necessarily hold.

"rss" is what's in physical memory;  "heapTotal" is memory reserved for the JS heap but it doesn't have to be in physical memory (yet - could be swapped out or simply not touched so far.)

**Comment by oancear05** [comment]
Do you know of any other similar situations with the heapTotal having an ascending trend? 
If yes, can you please tell me how much the memory will grow? Up to which value? 
Also, are you aware of the maximum value that heapTotal could achieve before the application stability is degraded?
Thank you!

**Comment by bnoordhuis** [comment]
The heap size is controlled by `--max_old_space_size=<mb>`. V8 allocates as needed up to that limit.

Application stability is a subjective measure, I don't really have an answer to that question.

I'll convert this to a discussion.

**Comment by gireeshpunathil** [comment]
- the `heapTotal` will be seen as ascending, in response to more object allocations in the app
 - the `heapTotal` may grow upto the maximum heap size set (either explicitly or the default)
 - the `heapTotal` and `rss` values have no direct relations. One may be lower, or higher than the other.
 - the `heapTotal` represents the managed memory (by the vm), so a perf degradation is not foreseen
